data_steward/cdr_cleaner/cleaning_rules/drop_multiple_measurements.py

```diff
@@ -10,89 +10,136 @@
-
-    valid_measurements_query = dict()
-    valid_measurements_query[cdr_consts.QUERY] = VALID_MEASUREMENTS.format(
-        dataset=dataset_id,
-        project=project_id,
-        sandbox_dataset=sandbox_dataset_id,
-        intermediary_table=INTERMEDIARY_TABLE)
-    queries_list.append(valid_measurements_query)
 
-    return queries_list
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'Removes all but the most recent of each Physical Measurement for all participants.'
+        super().__init__(issue_numbers=ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[cdr_consts.RDR],
+                         affected_tables=[MEASUREMENT],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_query_specs(self):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+
+        sandbox_invalid_rows = {
+            cdr_consts.QUERY:
+                SANDBOX_INVALID_MULT_MEASUREMENTS.render(
+                    project=self.project_id,
+                    dataset=self.dataset_id,
+                    sandbox_dataset=self.sandbox_dataset_id,
+                    intermediary_table=self.get_sandbox_tablenames()[0])
+        }
+
+        delete_invalid_rows = {
+            cdr_consts.QUERY:
+                REMOVE_INVALID_MULT_MEASUREMENTS.render(
+                    project=self.project_id,
+                    dataset=self.dataset_id,
+                    sandbox_dataset=self.sandbox_dataset_id,
+                    intermediary_table=self.get_sandbox_tablenames()[0])
+        }
+
+        return [sandbox_invalid_rows, delete_invalid_rows]
+
+    def setup_rule(self, client):
+        """
+        Function to run any data upload options before executing a query.
+        """
+        pass
+
+    def setup_validation(self, client):
+        """
+        Run required steps for validation setup
+        """
+        raise NotImplementedError("Please fix me.")
+
+    def validate_rule(self, client):
+        """
+        Validates the cleaning rule which deletes or updates the data from the tables
+        """
+        raise NotImplementedError("Please fix me.")
+
+    def get_sandbox_tablenames(self):
+        sandbox_table = f'{self._issue_numbers[0].lower()}_{self.affected_tables[0]}'
```

Comment:
I would suggest using `self.sandbox_table_for` defined in the base cleaning rule
```suggestion
        sandbox_table = self.sandbox_table_for(self.affected_tables[0])
```

Reply:
Updated!